base/shell/explorer/traywnd.cpp

```diff
@@ -2571,6 +2344,14 @@ class CTrayWindow :
         hRet = CTrayNotifyWnd_CreateInstance(m_hWnd, IID_PPV_ARG(IUnknown, &m_TrayNotifyInstance));
         if (FAILED_UNEXPECTEDLY(hRet))
             return FALSE;
+        IUnknown* tniUnknown = m_TrayNotifyInstance;
```

Comment:
Since this class also needs access to the button, perhaps we should create it here and pass it to the constructor of CTrayNotifyWnd